### ISSUE TITLE
[proposition] Add a env to the docker container to restore configs

### ISSUE THREAD

**Issue by cmoulliard** [fabric8io/fabric8#4031]
I'm thinking that we should provide a parameter (= env variable) to enable restore the configs when we mount the volumes for our continuous delivery docker containers (gerrit, gogs, jenkins, ...)

The new gerrit container will reuse the configs part of the volume mounted (https://github.com/christian-posta/ceposta-devops-ose/blob/master/gerrit-docker/src/main/docker/gerrit/bin/conf-and-run-gerrit.sh#L6) but we should provide a parameter to tell if we would like to restore the config or create a new setup

This proposition is valid for jenkins, gerrit & gogs (and could be extended for additional docker containers part of the CD scenario)

**Comment by jstrachan** [comment]
using real persistent volumes would do this right? persistent volumes are remounted when a pod is recreated

**Comment by cmoulliard** [comment]
Yes James but the idea of this variable is to tell to Docker when a new container is (re)created (gerrit, ..) that we will start with a fresh installation/setup and will not reuse what has been previously saved under the volume of the host (db, config files, ...). 

I have updated the title of the issue & comment to precise my intention.

For gerrit, we could add a supplementary test here by example with the env 'RESET=true/false" --> https://github.com/fabric8io/docker-gerrit/blob/master/bin/conf-and-run-gerrit.sh#L6-6

**Comment by jstrachan** [comment]
still not sure I grok what "restore configs" means and what you're trying to do. 

FWIW if you have 1 pod running and create a new pod via the RC, it'll create a new (empty) volume; you can then destroy the original pod. You then end up with a new, empty volume - if thats what "restore" means?

**Comment by cmoulliard** [comment]
I have changed the title to avoid confusion. This env variable `RESET` will be used when we create a new docker container (or that a pod recreates it) and where the volume mounted contains previously files created to tell to the shell script that we would like to do a reset = to initialise a new setup.

This is typically what is done when we run the first time the gerrit command ("https://github.com/fabric8io/docker-gerrit/blob/master/bin/conf-and-run-gerrit.sh#L11") but as such env variable don't exist, the only way that have for the moment to achieve a new init/setup of gerrit is to cleanup on the host the content of the directory mounted /home/gerrit-site.

**Comment by jstrachan** [comment]
I still don't grok how this would work. Env vars are fixed in a kube json. So to 'reset' you'd need to edit the json to enable the reset env var. So you'd have 2 app jsons; one with reset one without. The reset one would always reset on startup. Which sounds the same as using a non-persisent volume to me?

Otherwise resetting sounds like an operation you run on current pods; via a REST API or something?
